mobile/library/cc/engine_builder.h

```diff
@@ -66,8 +66,6 @@ class EngineBuilder {
   std::string generateConfigStr() const;
 
   EngineSharedPtr build();
-
-protected:
```

Comment:
I'm weary of making these two functions available to any user of the EngineBuilder class.  Can we keep it protected (or private) and make BaseClientIntegrationTest a friend class of EngineBuilder? https://en.cppreference.com/w/cpp/language/friend

Friendship is not inherited in C++, but we can get around that by adding functions to BaseClientIntegrationTest that call into the EngineBuilder's protected functions, and have ClientIntegrationTest and XdsIntegrationTest call into those.

Reply:
Learn something new about c++ every day!  Sounds good.